src/csharp/Record/Playback.cs

```diff
@@ -0,0 +1,786 @@
+                }
+
+                if (fileName == null)
+                {
+                    throw new ArgumentNullException(nameof(fileName));
+                }
+
+                // Determine the required buffer size
+                UIntPtr size = new UIntPtr(0);
+                if (NativeMethods.k4a_playback_get_attachment(this.handle, fileName, null, ref size) != NativeMethods.k4a_buffer_result_t.K4A_BUFFER_RESULT_TOO_SMALL)
+                {
+                    throw new AzureKinectException($"Unexpected internal state calling {nameof(NativeMethods.k4a_playback_get_track_name)}");
+                }
+
+                // Allocate a buffer
+                byte[] buffer = new byte[checked((int)size)];
+
+                // Get the codec id
+                AzureKinectGetTrackNameException.ThrowIfNotSuccess(() => NativeMethods.k4a_playback_get_attachment(this.handle, fileName, buffer, ref size));
+
+                return buffer;
+            }
+        }
+
+        /// <summary>
+        /// Read the next capture in the recording sequence.
+        /// </summary>
+        /// <returns>The next capture in the sequence, or null if at the end of the sequence.</returns>
+        /// <remarks>
+        /// <see cref="GetNextCapture"/> always returns the next capture in sequence after the most recently returned capture.
+        ///
+        /// The first call to <see cref="GetNextCapture"/> after <see cref="Seek(TimeSpan, PlaybackSeekOrigin)"/> will return the capture
+        /// in the recording closest to the seek time with an image timestamp greater than or equal to the seek time.
+        ///
+        /// If a call was made to <see cref="GetPreviousCapture"/> that returned null, the playback
+        /// position is at the beginning of the stream and <see cref="GetNextCapture"/> will return the first capture in the
+        /// recording.
+        ///
+        /// Capture objects returned by the playback API will always contain at least one image, but may have images missing if
+        /// frames were dropped in the original recording. When calling <see cref="Capture.Color"/>,
+        /// <see cref="Capture.Depth"/>, or <see cref="Capture.IR"/>, the image should be checked for null.
+        /// </remarks>
+        public Capture GetNextCapture()
+        {
+            lock (this)
+            {
+                if (this.disposedValue)
+                {
+                    throw new ObjectDisposedException(nameof(Playback));
+                }
+
+                IntPtr captureHandle = IntPtr.Zero;
+
+                switch (NativeMethods.k4a_playback_get_next_capture(this.handle, out captureHandle))
+                {
+                    case NativeMethods.k4a_stream_result_t.K4A_STREAM_RESULT_EOF:
+                        return null;
+                    case NativeMethods.k4a_stream_result_t.K4A_STREAM_RESULT_FAILED:
+                        throw new AzureKinectGetCaptureException();
```

Comment:
It might be nice to get the log here for this expcetion.

Reply:
And the other similar functions, GetPreviousCapture, GetNextImuSample, GetPreviousImuSample, etc